_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -413,12 +413,67 @@ public static function register_endpoints() {
 			'permission_callback' => __CLASS__ . '::view_admin_page_permission_check',
 		) );
 
-		// Plugins: get list of all plugins.
-		register_rest_route( 'jetpack/v4', '/plugins', array(
-			'methods' => WP_REST_Server::READABLE,
-			'callback' => __CLASS__ . '::get_plugins',
-			'permission_callback' => __CLASS__ . '::activate_plugins_permission_check',
-		) );
+		/*
+		 * Plugins: manage plugins on your site.
+		 *
+		 * @since 8.9.0
+		 *
+		 * @to-do: deprecate and switch to /wp/v2/plugins when WordPress 5.5 is the minimum required version.
+		 */
+		register_rest_route(
+			'jetpack/v4',
+			'/plugins',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => __CLASS__ . '::get_plugins',
+					'permission_callback' => __CLASS__ . '::activate_plugins_permission_check',
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => __CLASS__ . '::install_plugin',
+					'permission_callback' => __CLASS__ . '::activate_plugins_permission_check',
+					'args'                => array(
+						'slug'   => array(
+							'type'        => 'string',
+							'required'    => true,
+							'description' => __( 'WordPress.org plugin directory slug.', 'jetpack' ),
+							'pattern'     => '[\w\-]+',
+						),
+						'status' => array(
+							'description' => __( 'The plugin activation status.', 'jetpack' ),
```

Comment:
What do you think of setting up this endpoint to both install and activate a plugin if the status is `active`?

Reply:
That was my initial thought as well, especially since in our case you're rarely going to want to install a plugin but not activate it. Unfortunately, that's not how Core does it on their end right now:
https://github.com/WordPress/WordPress/blob/1f85e7484f154912d8869886ee3e5b68b06c0bc4/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php

If we want to eventually deprecate our custom endpoints in favor of core, I think we should try to stick to the way they do things from the start.

Reply:
Yep. This should be possible as of 7da961a391036fdae4d6c863924d60c013fa8a30